### PR TITLE
[@vimeo/vimeo] Add types for @vimeo/vimeo

### PR DESCRIPTION
**Pull request by MysteryBlokHed**
Related discussion: #69114

These types are copied with little modification from the `@types/vimeo` package, since the library was migrated to this new name without change.

If desired, I can add the old package's owner to this new one, since they actually created the types here.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/vimeo__vimeo/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 668463ee9f85c6e84d55582566ffbeeec47d6e2c
</p>

File: types/vimeo__vimeo/package.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    ],
+    "devDependencies": {
+        "@types/vimeo__vimeo": "workspace:."
+    },
+    "owners": [
+        {
+            "name": "Adam Thompson-Sharpe",
+            "githubUsername": "MysteryBlokHed"
+        }
+    ]
```
Comment (MysteryBlokHed) [inline types/vimeo__vimeo/package.json:16]: Like I mentioned in the PR body, I can add the original owner from the vimeo package here.
cc. @mattleff

**Review by peterblazejewicz — APPROVED** [review]
LGTM!


(note: do remove sources for `vimeo` once this one gets merged)

**Comment by MysteryBlokHed** [comment]
Ready to merge

**Comment by karlhorky** [comment]
> since the library was migrated to this new name without change.

This is not true - there is now a promise API, check my partial workaround / partial types here:

- https://github.com/vimeo/vimeo.js/issues/42#issuecomment-1657206137

**Pull request merged**